lib/swagger/diff/specification.rb

```diff
@@ -211,16 +220,15 @@ def hash_property(definition, prefix, name = '')
         "#{key} (in: body, type: Hash[string, #{type}])"
       end
 
-      # rubocop:disable Metrics/CyclomaticComplexity
       def properties(properties, required, prefix = '')
         ret = { required: Set.new, all: Set.new }
         properties.each do |name, schema|
           if schema['type'] == 'array'
             merge_refs!(ret, properties_for_ref(prefix, name, schema['items'], required, true))
-          elsif schema['type'] == 'object'
+          elsif schema['type'] == 'object' || schema.key?('properties')
             if schema['allOf']
               # TODO: handle nested allOfs.
-            elsif schema['type'] && schema['type'] == 'object' && schema['properties']
+            elsif schema['properties']
```

Comment:
is there a reason this is `schema['properties']` and a few lines above we use `schema.key?('properties')`?

Reply:
Nope, just written at two different times. I'll standardize on `schema['properties']` for robustness.